pyresample/geometry.py

```diff
@@ -1605,7 +1605,7 @@ def lonlat2colrow(self, lons, lats):
     def get_xy_from_lonlat(self, lon, lat):
         """Retrieve closest x and y coordinates.
 
-        Retrieve closest x and y coordinates (column, row indices) for the
+        Retrieve x and y coordinates (column, row indices) for the
```

Comment:
Is this no longer true? It is not the closest? I see you removed the astype for int. Does that mean the returned values are now floats instead of integers?

Reply:
It is not the closest anymore, it is the actual coordinates, as floats.

Reply:
Ok so people can't use this to directly index into their arrays? This is a pretty big backwards incompatibility. I've told people about this functionality at least twice in the last couple months. This also overlaps with #312.

Reply:
Yes, I know this is controversial. And I noted the overlap above. An alternative would be to have eg: `get_float_xy_from_lonlat` that does the heavy lifting and then 
```python
def get_xy_from_lonlat(self, lon, lat):
    return self.get_float_xy_from_lonlat(lon, lat).astype(int)
```
or something.

Reply:
> And I noted the overlap above

Ah I missed that, sorry. My gut reaction is to go with #312's approach of a keyword argument and have it default to integer.

Reply:
I personally don't like the addition of the flag to the function at all. I think my proposed approach keeps the code cleaner: less arguments, less if statements in the `get_xy_from_lontat` function for example.

Reply:
Fine by me. While we're at it, maybe call it something other than x/y? `rowcol`? I know there are some wrapper methods with similar naming.

Reply:
For me, row/col would connote integers, while x/y could be floats... in which case we would be backwards incompatible.

Reply:
Good point. For me x/y makes me assume projection-space meters. "indexes"? "indices"?

Reply:
indices is also ints in my world...
Other proposal: we deprecate the `get_xy_from_*` functions and create both ints and float functions instead.
We already have`lonlat2colrow`as an alias for `get_xy_from_lonlat`, so we could go for `lonlat2xy`.
Similarly, we could have `proj_coords2colrow` instead of `get_xy_from_proj_coords`, and `proj_coords2xy`.
As you mentioned, x/y already sounds like projection-space coordinates, but I can't fine an alternative atm...

Reply:
Maybe we ask twitter? We could make the distinction between coords (x/y projection coordinates), row/col (indexes), and y/x (fractional indexes)?

Reply:
For reference, after what we discussed on slack, I will implement the following:
- `get_array_indices_from_lonlat`, deprecating `get_xy_from_lonlat` and `lonlat2colrow`
- `get_array_coordinates_from_lonlat`, the floating point version
- `get_projection_coordinates_from_lonlat`
- `get_array_indices_from_projection_coordinates`, deprecating `get_xy_from_proj_coords`
- `get_array_coordinates_from_projection_coordinates`
- `get_lonlat_from_projection_coordinates`
- `get_lonlat_from_array_coordinates`, deprecating `get_lonlat` and `colrow2lonlat`, accepting both ints and floats.

Reply:
Those are really long names, but I'm not sure what we can do about that. All of these will support dask arrays too? And will be documented somewhere in the sphinx docs? :pray:

Reply:
> All of these will support dask arrays too?

What do you mean? that they will have a `chunks` argument? or that they work when `lons` and `lats` attributes are dask arrays?

Reply:
I have code that would break if `get_xy_...` would be changed to return floats instead of ints.

Reply:
I have preserved the ints.

Reply:
Yes it should
```suggestion
        Retrieve the closest x and y coordinates (column, row indices) for the
```